packages/api/src/auth/auth.controller.spec.ts

```diff
@@ -1,39 +1,91 @@
 /* eslint-disable no-underscore-dangle */
+import { JwtService } from '@nestjs/jwt';
 import { Test } from '@nestjs/testing';
 import httpMocks from 'node-mocks-http';
+import { v4 as uuidv4 } from 'uuid';
 
 import { env } from '@/common/env';
 
 import { AuthController } from './auth.controller';
+import { AuthService } from './auth.service';
+import type { LocalGuardRequest } from './local/local.types';
 
 describe('Auth controller', () => {
   let controller: AuthController;
+  let authService: Partial<AuthService>;
+  let jwtService: Partial<JwtService>;
+
+  const mockedJwt = 'some token value';
 
   beforeEach(async () => {
+    authService = {
+      register: jest.fn(),
+    };
+    jwtService = {
+      sign: jest.fn().mockReturnValue(mockedJwt),
+    };
+
     const module = await Test.createTestingModule({
       controllers: [AuthController],
+      providers: [
+        { provide: AuthService, useValue: authService },
+        { provide: JwtService, useValue: jwtService },
+      ],
     }).compile();
 
     controller = await module.resolve(AuthController);
   });
 
+  describe('register', () => {
+    it('calls `authService` with received body', async () => {
+      const bodyMock = { fullName: 'Some Name', email: 'some@mail.com', password: 'xyz' };
+
+      await controller.register(bodyMock);
+
+      expect(authService.register).toHaveBeenCalledWith(bodyMock);
```

Comment:
I'm not sure about this test. If service is mocked I'd rather define mocks behaviour in GIVEN, WHEN make a request, THEN assert with expect if controller response is conrrect based on GIVEN mocks.

Reply:
I'm not sure if I understand correctly. Do you want to remove current expects and replace them with `expect(response).toBe(undefined);`?

Reply:
Oh... hm... I haven't noticed that. But after what you said... I don't see value of this tests :P 
Maybe it shouldn't be discussed here, and leave testing for e2e.